presto-main/src/main/java/com/facebook/presto/operator/scalar/CharacterStringCasts.java

```diff
@@ -0,0 +1,198 @@
+    public static final SqlScalarFunction VARCHAR_TO_CHAR_CAST = trimCast(VARCHAR, CHAR);
+    public static final SqlScalarFunction CHAR_TO_VARCHAR_CAST = padSpacesCast(CHAR, VARCHAR);
+    public static final SqlScalarFunction VARCHAR_TO_CHAR_SATURATED_FLOOR_CAST = varcharToCharSaturatedFloorCast();
+
+    private CharacterStringCasts() {}
+
+    private static SqlScalarFunction truncateCast(String sourceType, String resultType)
+    {
+        Signature signature = getCastSignature(CAST, sourceType, resultType);
+        return SqlScalarFunction.builder(CharacterStringCasts.class)
+                .signature(signature)
+                .implementation(b -> b
+                        .methods("truncate")
+                        .withPredicate(context -> getArgumentTypeLength(context) > getReturnTypeLength(context))
+                        .withExtraParameters(context -> ImmutableList.of(getReturnTypeLength(context))))
+                .implementation(b -> b
+                        .methods("identity"))
+                .build();
+    }
+
+    private static SqlScalarFunction padSpacesCast(String sourceType, String resultType)
+    {
+        Signature signature = getCastSignature(CAST, sourceType, resultType);
+        return SqlScalarFunction.builder(CharacterStringCasts.class)
+                .signature(signature)
+                .implementation(b -> b
+                        .methods("padSpaces")
+                        .withExtraParameters(context -> ImmutableList.of(min(getArgumentTypeLength(context), getReturnTypeLength(context)))))
+                .build();
+    }
+
+    private static SqlScalarFunction trimCast(String sourceType, String resultType)
+    {
+        Signature signature = getCastSignature(CAST, sourceType, resultType);
+        return SqlScalarFunction.builder(CharacterStringCasts.class)
+                .signature(signature)
+                .implementation(b -> b
+                        .methods("trimSpaces")
+                        .withExtraParameters(context -> ImmutableList.of(getReturnTypeLength(context))))
+                .build();
+    }
+
+    private static SqlScalarFunction varcharToCharSaturatedFloorCast()
+    {
+        Signature signature = getCastSignature(SATURATED_FLOOR_CAST, VARCHAR, CHAR);
+        return SqlScalarFunction.builder(CharacterStringCasts.class)
+                .signature(signature)
+                .implementation(b -> b
+                        .methods("varcharToCharSaturatedFloorCast")
+                        .withExtraParameters(context -> ImmutableList.of(getReturnTypeLength(context))))
+                .build();
+    }
+
+    private static Signature getCastSignature(OperatorType operatorType, String sourceType, String resultType)
+    {
+        return Signature.builder()
+                .kind(SCALAR)
+                .operatorType(operatorType)
+                .typeVariableConstraints(withVariadicBound("X", sourceType), withVariadicBound("Y", resultType))
```

Comment:
Why variadic bound.
It should be sth like:
```
        return Signature.builder()
                .kind(SCALAR)
                .operatorType(operatorType)
                .argumentTypes(new TypeSignature(sourceType, TypeSignatureParameter.of("s")))
                .returnType(new TypeSignature(resultType, TypeSignatureParameter.of("r")))
                .build();
```

And then use `get*Length` function would look like:

```
    private static int getArgumentTypeLength(SpecializeContext context)
    {
        return context.getLiteral("s").intValue();
    }

    private static int getReturnTypeLength(SpecializeContext context)
    {
        return context.getLiteral("r").intValue();
    }
```

But as I said. All of that is not needed if you move to annotations.